libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestRequest.java

```diff
@@ -114,26 +143,96 @@ public String getPath(Object... args) {
+     */
+    public RestRequest(RestMethod method, String path) {
+        this(method, path, (RequestBody) null, null);
+    }
+
+    /**
+     * Generic constructor for arbitrary requests without a body.
+     *
+     * @param method				the HTTP method for the request (GET/POST/DELETE etc).
+     * @param path					the URI path, this will automatically be resolved against the users current instance host.
+     * @param additionalHttpHeaders additional headers.
+     *
+     */
+    public RestRequest(RestMethod method, String path,  Map<String, String> additionalHttpHeaders) {
+        this(method, path, (RequestBody) null, additionalHttpHeaders);
+    }
+
 	/**
 	 * Generic constructor for arbitrary requests.
 	 * 
-	 * @param method				the HTTP method for the request (GET/POST/DELETE etc)
+	 * @param method				the HTTP method for the request (GET/POST/DELETE etc).
 	 * @param path					the URI path, this will automatically be resolved against the users current instance host.
 	 * @param requestBody			the request body if there is one, can be null.
+     *
+     * Note: do not use this constructor if requestBody is not null and you want to build a batch or composite request.
 	 */
 	public RestRequest(RestMethod method, String path, RequestBody requestBody) {
-		this(method, path, requestBody, null);
+        this(method, path, requestBody, null);
 	}
 
-	public RestRequest(RestMethod method, String path, RequestBody requestBody, Map<String, String> additionalHttpHeaders) {
-		this.method = method;
-		this.path = path;
-		this.requestBody = requestBody;
-		this.additionalHttpHeaders = additionalHttpHeaders;
-	}
-	
-	/**
+    /**
+     * Generic constructor for arbitrary requests.
+     *
+     * @param method				the HTTP method for the request (GET/POST/DELETE etc).
+     * @param path					the URI path, this will automatically be resolved against the users current instance host.
+     * @param requestBodyAsJson		the request body as JSON if there is one, can be null.
+     *
+     * Note: use this constructor if requestBody is not null and you want to build a batch or composite request.
+     */
+    public RestRequest(RestMethod method, String path, JSONObject requestBodyAsJson) {
+        this(method, path, requestBodyAsJson, null);
+    }
+
+
+    /**
+     * Generic constructor for arbitrary requests.
+     *
+     * @param method				the HTTP method for the request (GET/POST/DELETE etc).
+     * @param path					the URI path, this will automatically be resolved against the users current instance host.
+     * @param requestBody			the request body if there is one, can be null.
+     * @param additionalHttpHeaders additional headers.
+     *
+     * Note: do not use this constructor if requestBody is not null and you want to build a batch or composite request.
+     */
+    public RestRequest(RestMethod method, String path, RequestBody requestBody, Map<String, String> additionalHttpHeaders) {
```

Comment:
This constructor takes a RequestBody which might or might not be JSON - so we live the requestBodyJson as null - a request built from this constructor won't be usable in a batch or composite request (there is a note about it in the method doc)